### DISCUSSION TITLE
Issues running AdGuard Home's DNSCrypt Server

### DISCUSSION THREAD

**Discussion by kikaragyozov** [AdguardTeam/AdGuardHome#7230] [Q&A]
The guide fails to mention after you've successfully configured a DNSCrypt server to run on port 5443, what exactly you have to do in order to get it to work.

For example, do I just copy the STAMP as an upstream DNS server for AdGuardHome, or will that cause some sort of a self-recursion? 

When I did that and deleted everything else, the response time for a single query sky-rocketed to 20,000 ms.

So something is definitely going wrong.

It also fails to mention what hostname should you use for the stamp - can it work with an internal IP, or does it have to be the external ip for which you've issued a valid TLS certificate?

**Comment by Dynamic5912** [comment]
> The guide fails to mention after you've successfully configured a DNSCrypt server to run on port 5443, what exactly you have to do in order to get it to work.
> 
> 
> 
> For example, do I just copy the STAMP as an upstream DNS server for AdGuardHome, or will that cause some sort of a self-recursion? 
> 
> 
> 
> When I did that and deleted everything else, the response time for a single query sky-rocketed to 20,000 ms.
> 
> 
> 
> So something is definitely going wrong.
> 
> 
> 
> It also fails to mention what hostname should you use for the stamp - can it work with an internal IP, or does it have to be the external ip for which you've issued a valid TLS certificate?

Configured what? 

AGH supports DNSCrypt out of the box for upstream DNS resolvers/providers.

All you need to do is add the upstream DNS you want to use.

**Reply by kikaragyozov** [reply]
As the title says - DNSCrypt server, not DNSCrypt upstream resolver.

See: https://github.com/AdguardTeam/AdGuardHome/wiki/DNSCrypt

I'm not sure if I'm supposed to add AdGuard Home's self-hosted DNSCrypt server as upstream. When I did, query time was around 20,000 ms, which is 20 seconds. Completely unusable.

**Comment by kikaragyozov** [comment]
Is there anyone capable of answering this?
